app/src/lib/stores/app-store.ts

```diff
@@ -1771,6 +1773,40 @@ export class AppStore extends TypedBaseStore<IAppState> {
     this._updateCurrentPullRequest(repository)
     this.updateMenuItemLabels(repository)
     this._initializeCompare(repository)
+    this.repositories = await this.getRefreshedRepositories(
+      this.repositories,
+      repository
+    )
+  }
+
+  public async refreshAllRepositories() {
+    this.repositories = await this.getRefreshedRepositories(this.repositories)
+    this.emitUpdate()
+  }
+
+  private async getRefreshedRepositories(
+    repositories: ReadonlyArray<Repository>,
+    targetRepo?: Repository
+  ): Promise<ReadonlyArray<Repository>> {
```

Comment:
I think we can make this a `Promise<void>`-returning function and update the name now that we're updating a cache based on the repository ID, which will eliminate a bunch of the `this.repositories` assignment.

Reply:
Also, for functions that are `Promise<void>`-returning we try and structure them so they manage their own `this.emitUpdate()` responsibility. An example of that is `updateRepositorySelectionAfterRepositoriesChanged()` and the check at the end:

```ts
    if (repositoryChanged) {
      this._selectRepository(newSelectedRepository)
      this.emitUpdate()
    }
```

It'd be nice to have this function behave in the same way, so it only signals when it's cache has been added or updated.